ee/hogai/graph/memory/prompts.py

```diff
@@ -111,23 +111,30 @@
 
 <instructions>
 When new information is provided, follow these steps:
-1. Analyze the information:
+1. Check if the user is explicitly asking you to remember something:
+   - If the user's message contains phrases like "remember that", "remember this", "save this", "note that", etc., you MUST save the information regardless of whether it's product-related.
+   - Extract the content after the explicit request and save it using 'core_memory_append'.
+2. If not an explicit request, analyze the information:
    - Determine if the information is relevant and which memory type it belongs to.
    - If relevant, formulate a clear, factual statement based on the information.
    - Consider the implications of this new information on existing memory.
    - Decide whether to append this new information or replace existing information in the core memory, providing reasoning for your decision.
    - Keep reasoning short and concise under 50 words.
-2. If relevant, update the core memory using the 'core_memory_append' or 'core_memory_replace' tool as appropriate. To update the memory you MUST use either of these tools.
-3. Output "[Done]" when you have finished processing the information. IMPORTANT: If the input does not contain new product-related information, return "[Done]" without any explanation.
+3. If relevant or explicitly requested, update the core memory using the 'core_memory_append' or 'core_memory_replace' tool as appropriate. To update the memory you MUST use either of these tools.
+4. Output "[Done]" when you have finished processing the information. IMPORTANT: If the input does not contain new product-related information AND the user did not explicitly request to save it, return "[Done]" without any explanation.
 
 Ignore phrases that:
 - Are too vague or generic without specific details (e.g., "pageview trend").
 - Do not describe actions, attributes, or implications related to the company or product.
+- EXCEPTION: Always save information when explicitly requested by the user, even if vague or not product-related.
 </instructions>
 
 <examples>
 Here are some few shot examples:
 
+Input: Remember my favorite treat is chocolate
+Output: The user's favorite treat is chocolate. [Call core_memory_append with "The user's favorite treat is chocolate."]
```

Comment:
Perhaps we just need to add a Reasoning block here.
```suggestion
Output: The user's favorite treat is chocolate.
Reasoning: The user explicitly asked to save it.
```